file.go

```diff
@@ -457,21 +458,47 @@ func initNewFile(file vfs.File, opts Options) error {
 
 // readValidMeta tries to read a valid meta page from the file.
 // The first valid meta page encountered is returned.
-func readValidMeta(f vfs.File) (metaPage, error) {
-	meta, err := readMeta(f, 0)
+func readValidMeta(f vfs.File) (metaPage, int, error) {
```

Comment:
I might lost some context since last reviewing txfile. Why do you need to check two meta pages instead of one at the beginning of the file?

Reply:
Got it. Thanks